cli/azd/pkg/infra/provisioning/provider.go

```diff
@@ -70,23 +76,29 @@ type Provider interface {
 	Destroy(ctx context.Context, deployment *Deployment, options DestroyOptions) *async.InteractiveTaskWithProgress[*DestroyResult, *DestroyProgress]
 }
 
+// Registers a provider creation function for the specified provider kind
+func RegisterProvider(kind ProviderKind, newFn NewProviderFn) error {
+	providers[kind] = newFn
+
+	return nil
+}
+
 func NewProvider(ctx context.Context, env *environment.Environment, projectPath string, infraOptions Options) (Provider, error) {
 	var provider Provider
 
-	switch infraOptions.Provider {
-	case Bicep:
-		provider = NewBicepProvider(ctx, env, projectPath, infraOptions)
-	case Test:
-		provider = NewTestProvider(ctx, env, projectPath, infraOptions)
-	default:
-		provider = NewBicepProvider(ctx, env, projectPath, infraOptions)
+	if infraOptions.Provider == "" {
+		infraOptions.Provider = Bicep
 	}
 
-	if provider != nil {
-		return provider, nil
+	newProviderFn, ok := providers[infraOptions.Provider]
+	if !ok {
+		return nil, fmt.Errorf("provider '%s' is not supported", infraOptions.Provider)
 	}
 
-	return nil, fmt.Errorf("provider '%s' is not supported", infraOptions.Provider)
-}
+	provider, err := newProviderFn(ctx, env, projectPath, infraOptions)
+	if err != nil {
+		return nil, fmt.Errorf("error creating provider for type '%s'", infraOptions.Provider)
```

Comment:
Do you want to include the `error` from `newProviderFn`?

I'm guessing it would be useful if `newProviderFn` provided validation of sorts